File: Library/Library.Core/Models/Book.cs
using Microsoft.AspNetCore.Http;

namespace Library.Core.Models
{
    public class Book
    {
        public Guid Id { get; set; }
        public string ISBN { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Genre { get; set; }
        public DateTime TakenAt { get; set; }
        public DateTime ReturnBy { get; set; }
        public IFormFile ImageFile { get; set; }
        public string ImagePath { get; set; }
        public ICollection<Author> Authors { get; set; } = [];
    }
}

File: library-frontend/src/app/core/models/book.ts
import { Guid } from "guid-typescript";
import { Author } from "./author";

export interface Book {
    id: Guid;
    name: string;
    description: string;
    genre: string;
    isbn: string;
    returnBy?: Date;
    takenAt?: Date;
    imagePath?: string;

    authors: Author[];
}